include/deal.II/grid/tria_accessor.h

```diff
@@ -3240,7 +3240,12 @@ class CellAccessor : public TriaAccessor<dim, dim, spacedim>
    */
   boost::container::small_vector<
     TriaIterator<TriaAccessor<dim - 1, dim, spacedim>>,
-    GeometryInfo<dim>::faces_per_cell>
+#ifndef _MSC_VER // MSVC prior to 2022 cannot use a constexpr function this way
+    ReferenceCell::max_n_faces<dim>()
+#else
+    GeometryInfo<dim>::faces_per_cell
+#endif
+    >
```

Comment:
I think we should upgrade the minimum version of MSVC soon so we can do this renaming without writing a ton of these preprocessor checks.